### DISCUSSION TITLE
Combine connectOrCreate and upsert?

### DISCUSSION THREAD

**Discussion by janpio** [prisma/prisma#6632] [Q&A]
I naively wrote this query:

```
const upsert = await prisma.lambda.upsert({
  where: { lambdaid: lambdaid },
  create: {
    lambdaid: lambdaid,
    invocation_count: 1,
    testrun: {
      upsert: {
        where: { name: testrun },
        create: { name: testrun, count: 1 },
        update: { count: { increment: 1 }},
      }
    }
  },
  update: {
    invocation_count: {
      increment: 1
    }
  }
})
```

I tried to upsert a `lambda` and connect it to a new or existing `testrun` in the `create` case, and update the existing counter in that `testrun` by `+1` as well - but I get this error:

```
Error: 
Invalid `prisma.lambda.upsert()` invocation:

{
  where: {
    lambdaid: 'fcd0d2eb580a3e5be9aae33f80673508'
  },
  create: {
    lambdaid: 'fcd0d2eb580a3e5be9aae33f80673508',
    invocation_count: 1,
    testrun: {
      upsert: {
      ~~~~~~
        where: {
          name: 'foo'
        },
        create: {
          name: 'foo',
          count: 1
        },
        update: {
          count: {
            increment: 1
          }
        }
      }
    }
  },
  update: {
    invocation_count: {
      increment: 1
    }
  }
}

Unknown arg `upsert` in create.testrun.upsert for type testrunCreateNestedOneWithoutLambdasInput. Did you mean `select`? Available args:
type testrunCreateNestedOneWithoutLambdasInput {
  create?: testrunCreateWithoutLambdasInput | testrunUncheckedCreateWithoutLambdasInput
  connectOrCreate?: testrunCreateOrConnectWithoutLambdasInput
  connect?: testrunWhereUniqueInput
}
```

So seems I should use `connectOrCreate` here instead, as this is a relation.

Can I somehow update the `testrun.count` in this one query via `connectOrCreate` anyway?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
